fortinet-fortiflex/fortiflex_api_auth.py

```diff
@@ -0,0 +1,139 @@
+""" Copyright start
+  Copyright (C) 2008 - 2023 Fortinet Inc.
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+import requests, json
+from time import time, ctime
+from datetime import datetime
+from connectors.core.utils import update_connnector_config
+from connectors.core.connector import get_logger, ConnectorError
+
+logger = get_logger('fortinet-fortiflex')
+
+REFRESH_TOKEN_FLAG = False
+REFRESH_TOKEN = 'refresh_token'
+
+
+class FortiFlex:
+    def __init__(self, config):
+        self.username = config.get("username")
+        self.password = config.get("password")
+        self.client_id = config.get("client_id")
+        self.verify_ssl = config.get('verify_ssl')
+        self.host = config.get("server")
+        if self.host[:7] == "http://":
+            self.host = "https://{0}".format(self.host)
+        elif self.host[:8] == "https://":
+            self.host = "{0}".format(self.host)
+        else:
+            self.host = "https://{0}".format(self.host)
+
+    def convert_ts_epoch(self, ts):
+        datetime_object = datetime.strptime(ctime(ts), "%a %b %d %H:%M:%S %Y")
+        return datetime_object.timestamp()
+
+    def generate_token(self, REFRESH_TOKEN_FLAG):
```

Comment:
REFRESH_TOKEN_FLAG can be used directly instead of taking as function argument, as the value will be same everytime if it is used this way

Reply:
When we are generating token first time, flag will be False, but after validating the token in next iteration's it change as True, we are not keeping same as every time.